cprover-cpp-api/src/api.cpp

```diff
@@ -0,0 +1,52 @@
+#include "api.h"
+
+#include <util/cmdline.h>
+#include <util/config.h>
+#include <util/message.h>
+#include <util/options.h>
+
+#include <goto-programs/goto_model.h>
+#include <goto-programs/initialize_goto_model.h>
+
+#include <ansi-c/ansi_c_language.h>
+#include <langapi/mode.h>
+
+#include <memory>
+#include <vector>
+#include <string>
+
+extern configt config;
+
+cprover_api::cprover_api()
+  : message_handler(
+      util_make_unique<null_message_handlert>(null_message_handlert{})),
+    options(util_make_unique<optionst>(optionst{}))
+{
+  // Needed to initialise the language options correctly
+  cmdlinet cmdline;
+  // config is global in config.cpp
+  config.set(cmdline);
+  // Initialise C language mode
+  register_language(new_ansi_c_language);
```

Comment:
For a first implementation this is ok I think, but do we want to force/hard-code that the language is C?

Reply:
This is indeed not ideal, but for a first implementation we need to have at least one language mode bootstrapped, and none other than `C` seem to fit in this particular scenario.

What I can do is potentially add a `todo` comment to make this configurable in the future.